connect-location/src/main/java/com/ifttt/location/ConnectLocation.java

```diff
@@ -10,19 +15,29 @@
 import com.ifttt.connect.api.Connection;
 import com.ifttt.connect.api.ConnectionApiClient;
 import com.ifttt.connect.api.ErrorResponse;
+import com.ifttt.connect.api.PendingResult;
 import com.ifttt.connect.api.UserTokenProvider;
 import com.ifttt.connect.ui.ButtonStateChangeListener;
 import com.ifttt.connect.ui.ConnectButton;
 import com.ifttt.connect.ui.ConnectButtonState;
+import com.ifttt.connect.ui.PendingResultLifecycleObserver;
 import java.util.concurrent.TimeUnit;
 
+/**
+ * The main class for the Connect Location SDK. This class handles state change events from {@link ConnectButton},
+ * sets up polling to get the latest location field values set for the connection, and
+ * provides a function to add a listener to be notified when location permissions are required from the user
+ */
 public final class ConnectLocation implements ButtonStateChangeListener {
 
     private static ConnectLocation INSTANCE = null;
     final GeofenceProvider geofenceProvider;
     final ConnectionApiClient connectionApiClient;
     private final WorkManager workManager;
 
+    private final LocationPermissionChecker permissionChecker;
+    private LocationPermissionCallback permissionCallback;
```

Comment:
If this is nullable, we should use `@Nullable` annotation. This package's fields are non-null by default.